apps/sms/views/shared/js/activity_handler.js

```diff
@@ -21,22 +23,27 @@ const ActivityDataType = {
 };
 
 var ActivityHandler = {
-  // Will hold current activity object
-  _activity: null,
+  _activityClient: null,
 
   init: function() {
     if (!window.navigator.mozSetMessageHandler) {
       return;
     }
 
-    // A mapping of MozActivity names to their associated event handler
-    window.navigator.mozSetMessageHandler('activity',
-      this._onActivity.bind(this, {
-        'new': this._onNewActivity,
-        'share': this._onShareActivity
-      })
+    this._activityClient = new ActivityClient();
+
+    this._activityClient.on(
+      'new-activity-request', this._onNewActivity.bind(this)
+    );
+
+    this._activityClient.on(
+      'share-activity-request', this._onShareActivity.bind(this)
     );
 
+    // Currently both client and service for activity resides in the same
+    // context, so we instantiate both.
+    (new ActivityService());
```

Comment:
sorry, my comment was not related to your comment, but it was related to the same line of code :)